### PR TITLE
Merged HDI + some convenient shortcuts for engines + a few more minor things

### PR DESCRIPTION
**Pull request by alexandrebouchard**

### CODE REVIEW

**Comment by alexandrebouchard** [comment]
Hey Kevin, once the new version of HDI is pushed, could you merge with these commits? Thanks!

**Comment by kevinchern** [comment]
Still investigating. For some reason testNormalizationEstimates is passing locally :+1:  but failing on travis.

**Comment by kevinchern** [comment]
Got it!! Will make a push soon.

**Comment by alexandrebouchard** [comment]
Awesome! What made the travis test crash?

**Comment by kevinchern** [comment]
> Awesome! What made the travis test crash?

If https://github.com/UBC-Stat-ML/blangSDK/pull/162 passes tests on travis, it's because the
```
if (previous !== null)
  assertEquals( ... )
```
was buggy. Eclipse didn't catch it for some reason...

File: src/main/java/blang/engines/internals/factories/SCM.java [hunk 1]
```diff
@@ -48,7 +48,10 @@ public void performInference()
     // write Z estimate
     double logNormEstimate = approximation.logNormEstimate();
     System.out.println("Log normalization constant estimate: " + logNormEstimate);
-    BriefIO.write(results.getFileInResultFolder(Runner.LOG_NORM_ESTIMATE), "" + logNormEstimate);
+    results.getTabularWriter(Runner.LOG_NORMALIZATION_ESTIMATE).write(
+        Pair.of(Runner.LOG_NORMALIZATION_ESTIMATOR, "SMC"),
```
[the remark this answers is not in the document]
Comment (kevinchern) [reply src/main/java/blang/engines/internals/factories/SCM.java:52]: if yes to both i'll change and merge :+1:

File: src/main/java/blang/engines/internals/factories/PT.java [hunk 2]
```diff
@@ -313,7 +314,10 @@ private void informedInitialization(SampledModel model)
         }
         double logNormEstimate = population.logNormEstimate();
         System.out.println("Log normalization constant estimate: " + logNormEstimate);
-        BriefIO.write(scmInit.results.getFileInResultFolder(Runner.LOG_NORM_ESTIMATE), "" + logNormEstimate);
+        results.getTabularWriter(Runner.LOG_NORMALIZATION_ESTIMATE).write(
+          Pair.of(Runner.LOG_NORMALIZATION_ESTIMATOR, "SMC-initialization"),
```
[the remark this answers is not in the document]
Comment (alexandrebouchard) [reply src/main/java/blang/engines/internals/factories/PT.java:318]: Good idea to make this change for consistency.

**Review by kevinchern — APPROVED** [review]
renamed 'smc' to 'scm' for consistency. will merge now.

**Pull request merged**